iotdb-core/datanode/src/main/java/org/apache/iotdb/db/utils/datastructure/TVList.java

```diff
@@ -321,17 +476,26 @@ protected long[] cloneTime(long[] array) {
     return cloneArray;
   }
 
-  void updateMaxTimeAndSorted(long[] time, int start, int end) {
+  void updateMinMaxTimeAndSorted(long[] time, int start, int end) {
     int length = time.length;
     long inPutMinTime = Long.MAX_VALUE;
     boolean inputSorted = true;
+    int inputSeqRowCount = 0;
     for (int i = start; i < end; i++) {
       inPutMinTime = Math.min(inPutMinTime, time[i]);
       maxTime = Math.max(maxTime, time[i]);
-      if (inputSorted && i < length - 1 && time[i] > time[i + 1]) {
-        inputSorted = false;
+      minTime = Math.min(minTime, time[i]);
+      if (inputSorted) {
+        if (i < length - 1 && time[i] > time[i + 1]) {
+          inputSorted = false;
+        } else {
+          inputSeqRowCount++;
+        }
       }
     }
+    if (sorted && (rowCount == 0 || time[start] > getTime(rowCount - 1))) {
+      seqRowCount += inputSeqRowCount;
+    }
```

Comment:
I think if time[start] == getTime(rowCount -1), the list is still sorted.

Reply:
yes

Reply:
Done